test/Array/Array_TypeConfusion_bugs.js

```diff
@@ -768,7 +768,7 @@ var tests = [
             arr.__proto__ = Array.prototype;
 
             assert.isTrue(getterCalled);
-            assert.areEqual([101,11,22,,77,16], arr, '77 and 16 are not part of the sort so they are not sorted');
+            assert.areEqual([101,11,101,,77,16], arr, '77 and 16 are not part of the sort so they are not sorted');

```

Comment:
This test case involves a getter which provides a value each time it's called but does not set anything back. The order of comparisons in your sort therefore effects what values end up in your sorted array - hence the change of output.

The test was meant to check that 77 and 16 which are added to the end of the array by the getter aren't sorted - that is still the case.